packages/web3/src/signer/tx-builder.ts

```diff
@@ -62,28 +68,126 @@ export abstract class TransactionBuilder {
     params: SignTransferTxParams,
     publicKey: string
   ): Promise<Omit<SignTransferTxResult, 'signature'>> {
+    const data = this.buildTransferTxParams(params, publicKey)
+    const response = await this.nodeProvider.transactions.postTransactionsBuild(data)
+    return { ...response, gasPrice: fromApiNumber256(response.gasPrice) }
+  }
+
+  async buildDeployContractTx(
+    params: SignDeployContractTxParams,
+    publicKey: string
+  ): Promise<Omit<SignDeployContractTxResult, 'signature'>> {
+    const data = this.buildDeployContractTxParams(params, publicKey)
+    const response = await this.nodeProvider.contracts.postContractsUnsignedTxDeployContract(data)
+    const contractId = binToHex(contractIdFromAddress(response.contractAddress))
+    return { ...response, groupIndex: response.fromGroup, contractId, gasPrice: fromApiNumber256(response.gasPrice) }
+  }
+
+  async buildExecuteScriptTx(
+    params: SignExecuteScriptTxParams,
+    publicKey: string
+  ): Promise<Omit<SignExecuteScriptTxResult, 'signature'>> {
+    const data = this.buildExecuteScriptTxParams(params, publicKey)
+    const response = await this.nodeProvider.contracts.postContractsUnsignedTxExecuteScript(data)
+    return { ...response, groupIndex: response.fromGroup, gasPrice: fromApiNumber256(response.gasPrice) }
+  }
+
+  async buildChainedTx(params: BuildChainedTxParams[]): Promise<BuildChainedTxResult[]> {
```

Comment:
Probably more consistent and convenient to provide a list of public keys than to moving public keys to params.

```
async buildChainedTx(params: BuildChainedTxParams[], publicKeys: string[]): ...
```